wpilibNewCommands/src/main/native/cpp/frc2/command/CommandScheduler.cpp

```diff
@@ -193,10 +193,22 @@ void CommandScheduler::Run() {
   m_watchdog.AddEpoch("buttons.Run()");
 
   m_impl->inRunLoop = true;
+  bool isDisabled = frc::RobotState::IsDisabled();
   // Run scheduled commands, remove finished commands.
   for (Command* command : m_impl->scheduledCommands) {
-    if (!command->RunsWhenDisabled() && frc::RobotState::IsDisabled()) {
-      Cancel(command);
+    if (!command->RunsWhenDisabled() && isDisabled) {
+      m_impl->ending.insert(command);
+      command->End(true);
+      m_impl->ending.erase(command);
+      for (auto&& action : m_impl->interruptActions) {
+        action(*command);
+      }
+
+      m_impl->scheduledCommands.erase(command);
+      for (auto&& requirement : command->GetRequirements()) {
+        m_impl->requirements.erase(requirement);
+      }
+      m_watchdog.AddEpoch(command->GetName() + ".End(true)");
```

Comment:
What's the rationale for this change / inlining of `cancel()`?
This also changes behavior: canceled commands will be removed immediately rather than being deferred to the end of the loop.

Reply:
I think the code deduplication and consistency is worth the redundant checks.